static/app/views/organizationIntegrations/integrationExternalTeamMappings.tsx

```diff
@@ -83,83 +86,82 @@ class IntegrationExternalTeamMappings extends AsyncComponent<Props, State> {
     return externalTeamMappings.sort((a, b) => parseInt(a.id, 10) - parseInt(b.id, 10));
   }
 
-  sentryNamesMapper(teams: Team[]) {
-    return teams.map(({id, slug}) => ({id, name: slug}));
-  }
+  modalMappingKey = 'MODAL_RESULTS';
 
-  handleSubmit = (
-    data: Record<string, any>,
-    onSubmitSuccess: (data: Record<string, any>) => void,
-    onSubmitError: (error: any) => void,
-    _: React.FormEvent<Element>,
-    model: FormModel,
-    mapping?: ExternalActorMapping
-  ) => {
-    // We need to dynamically set the endpoint bc it requires the slug of the selected team in the form.
-    try {
-      const {organization} = this.props;
-      const {queryResults} = this.state;
-      const team = queryResults.find(item => item.id === data.teamId);
+  get dataEndpoint() {
+    const {organization} = this.props;
+    return `/organizations/${organization.slug}/teams/`;
+  }
 
-      if (!team) {
-        throw new Error('Cannot find team slug.');
-      }
+  getBaseFormEndpoint(mapping?: ExternalActorMapping) {
```

Comment:
The above comment explains a lot but just to reiterate, this is where we use the results stored in state to get the team slug that we want to submit to. The endpoint is annoying dynamic for teams and is one of the reasons i'm on Team GraphQL.